unit-tests/repository_test.py

```diff
@@ -74,7 +74,21 @@ def test_extract_from_svn(self, repo: Tuple[Path, Path], tmp_path: Path):
         process = Popen(command)
 
         test_unit = SubversionRepo('svn://localhost/trunk')
-        test_unit.extract(tmp_path)
+        #
+        # It seems there can be a delay between the server starting and the
+        # listen socket opening. Thus we have a number of retries.
+        #
+        # TODO: Is there a better solution such that we don't try to connect
+        #       until the socket is open?
+        #
+        for retry in range(3, 0, -1):
+            try:
+                test_unit.extract(tmp_path)
+            except FabException as ex:
+                if range == 0:
+                    raise ex
```

Comment:
I think if you are re-raising the original exception you might be able to leave out the exception and just put `raise` - but at the same time it doesn't hurt anything to be explicit